<p>All $refs must reference a valid location in the document.<br/>
    Make sure you declare missing definitions or remove any unintentionally used references.
</p>

<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    post:
      parameters:
        - $ref: "#/parameters/Missing"           # Noncompliant {{Missing parameter}}
      responses:
        default:
          description: the default response
          schema:
            $ref: "#/definitions/Missing"        # Noncompliant {{Missing schema}}
        '201':
          $ref: "#/responses/Used"
    put:
      parameters:
        - $ref: "#/parameters/Used"
      responses:
        default:
          description: the default response
          schema:
            $ref: "#/definitions/Missing"        # Noncompliant {{Missing schema}}
        '201':
          $ref: "#/responses/Missing"            # Noncompliant {{Missing response}}
definitions:
  Used:
    type: object
parameters:
  Used:
    type: string
    in: query
    name: toto
responses:
  Used:
    description: some referenced response
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /pets:
    post:
      parameters:
        - $ref: "#/parameters/Missing"
      responses:
        default:
          description: the default response
          schema:
            $ref: "#/definitions/Missing"
        '201':
          $ref: "#/responses/Used"
    put:
      parameters:
        - $ref: "#/parameters/Used"
      responses:
        default:
          description: the default response
          schema:
            $ref: "#/definitions/Missing"
        '201':
          $ref: "#/responses/Missing"
definitions:
  Used:
    type: object
  Missing:
    type: object
parameters:
  Used:
    type: string
    in: query
    name: toto
  Missing:
    type: string
    in: query
    name: toto
responses:
  Used:
    description: some referenced response
  Missing:
    description: some referenced response
</pre>
